pkg/kube/client/ssa/gvk_parser.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+This file is forked from upstream kubernetes/apimachinery
+https://github.com/kubernetes/apimachinery/blob/2465dc5239ab8827a637148a78b380c278b4a5f4/pkg/util/managedfields/gvkparser.go
```

Comment:
why is it forked? What differs?

Reply:
was forked due to build a new parser without the proto.Models (as building a proto.Models from v3 doc had an upstream issue).
but realized that the only difference we introduce is about NewGVKParser, no need to redefine the struct itself. Will change 👍